### PR TITLE
WIP - add feature to "downscale" timeseries data to subregions

### PR DESCRIPTION
**Pull request by danielhuppmann**
# Please confirm that this PR has done the following:

- [x] Tests Added
- [x] Documentation Added
- [ ] Description in RELEASE_NOTES.md Added

## Adding to RELEASE_NOTES.md (remove section after adding to RELEASE_NOTES.md)

Please add a single line in the release notes similar to the following:

```
- (#XX)[http://link-to-pr.com] Added feature which does something
```

# Description of PR

This PR adds a feature to "downscale" timeseries data (one or multiple variables) from a "region" to a number of "subregions" using another variable as weight.

The PR is directed into `IAMconsortium:feature/aggregate_be_list` to highlight the differences of the aggregation-features to that prep-work - I'll rebase after a first round of review and merging of #306.

### CODE REVIEW

**Comment by gidden** [comment]
restarted ci (only linux py37)

File: tests/test_feature_downscale.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import pytest
+import pandas as pd
+import pyam
+
+
+@pytest.mark.parametrize("variable", (
+    ('Primary Energy'),
+    (['Primary Energy', 'Primary Energy|Coal']),
+))
+def test_downscale_region(aggregate_df, variable):
```
Comment (gidden) [inline tests/test_feature_downscale.py]: just to make sure, this test is doing the following, correct?

1. downscale regional values of a global variable by population
2. pull out existing regional values
3. confirm they are equal

Does that mean that the population weights were specifically chosen to match the regional breakdown? Smart =)

**Comment by gidden** [comment]
Is it possible to add something to the tutorials here? If not, can we add an issue for this? Perhaps a nice exercise for fabio to get started?

**Comment by danielhuppmann** [comment]
I'll include this in the updated aggregation tutorial

File: tests/test_feature_downscale.py (hunk 1 above)
Comment (danielhuppmann) [reply tests/test_feature_downscale.py]: > Smart =)

Thank you (and yes, I played around with the test data to cover aggregation (one simple aggregation, one example with weights, one with region-level components) and downscaling both ways...

**Comment by danielhuppmann** [comment]
closing in favour of #313

**Pull request closed without merging**